pkg/resources/resources.go

```diff
@@ -29,7 +27,7 @@ func ToUnstructured(resource metav1.Object) (*unstructured.Unstructured, error)
 }
 
 // ToProper returns the proper object representation of a resource.
-func ToProper(destination metav1.Object, source metav1.Object) error {
+func ToProper(destination, source client.Object) error {
```

Comment:
```suggestion
func ToTyped(destination, source client.Object) error {
```
If suggestion accepted, we'll have to make the relevant changes in operator-builder.